src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -332,9 +323,26 @@ private static void RecordRequestStartEventLog(HttpContext httpContext)
             out var requestId,
             out var traceState);
 
+        Activity? activity = _activitySource.CreateActivity(
+            ActivityName,
+            ActivityKind.Server,
+            string.IsNullOrEmpty(requestId) ? null! : requestId
+            );
+
+        if (activity is null)
```

Comment:
Weird that CreateActivity sometimes doesn't create an activity. Add a comment saying why activity might be null

Reply:
its a perf optimization, if there are no listeners the activity isn't created